Model/ResourceModel/NodeType/Product.php

```diff
@@ -0,0 +1,129 @@
+
+        $select = $connection->select()->from(
+            ['e' => $this->getTable('catalog_product_entity')],
+            [$identifierField, 'sku']
+        );
+
+        return $connection->fetchAll($select);
+    }
+
+    /**
+     * @param int   $storeId
+     * @param array $productIds
+     *
+     * @return array
+     */
+    public function fetchData($storeId = Store::DEFAULT_STORE_ID, $productIds = [])
+    {
+        $connection = $this->getConnection('read');
+        $table = $this->getTable('url_rewrite');
+        $select = $connection
+            ->select()
+            ->from($table, ['entity_id', 'request_path'])
+            ->where('entity_type = ?', 'product')
+            ->where('redirect_type = ?', 0)
+            ->where('store_id = ?', $storeId)
+            ->where('entity_id IN (' . implode(',', $productIds) . ')');
+
+        return $connection->fetchPairs($select);
+    }
+
+    /**
+     * @param int $websiteId
+     * @param int $customerGroupId
+     * @param array $productIds
+     * @return array
+     */
+    public function fetchPriceData($websiteId, $customerGroupId, $productIds = [])
+    {
+        $connection = $this->getConnection('read');
+        $table = $this->getTable('catalog_product_index_price');
+        $select = $connection
+            ->select()
+            ->from($table, ['entity_id', 'final_price'])
+            ->where('customer_group_id = ?', $customerGroupId)
+            ->where('website_id = ?', $websiteId)
+            ->where('entity_id IN (' . implode(',', $productIds) . ')');
+
+        return $connection->fetchPairs($select);
+    }
+
+    /**
+     * @param array $productIds
+     * @return array
+     */
+    public function fetchImageData($productIds = [])
+    {
+        $metadata = $this->metadataPool->getMetadata(ProductInterface::class);
+        $linkField = $metadata->getLinkField();
+
+        $connection = $this->getConnection('read');
+        $nameAttributeId = $this->getAttributeIdByCode($connection, 'image');
+
+        $table = $this->getTable('catalog_product_entity_varchar');
+        $select = $connection->select()
+            ->from($table, [$linkField, 'value'])
+            ->where('attribute_id = ?', $nameAttributeId)
+            ->where('store_id = ?', 0)
+            ->where($linkField .' IN (' . implode(',', $productIds) . ')');
+
+        return $connection->fetchPairs($select);
+    }
+
+    /**
+     * @param AdapterInterface $connection
+     * @param string $code
+     * @return array|string
+     */
+    private function getAttributeIdByCode($connection, $code)
+    {
+        $select = $connection->select()->from(
+            ['a' => $this->getTable('eav_attribute')],
+            ['attribute_id']
+        )->join(
+            ['t' => $this->getTable('eav_entity_type')],
+            't.entity_type_id = a.entity_type_id',
+            []
+        )->where('t.entity_type_code = ?', CoreProduct::ENTITY)->where(
```

Comment:
Please move the 2nd `where` to the next line (I think we can reformat all of this to keep the same structure as with `join`